tests/func/test_diff.py

```diff
@@ -264,3 +264,127 @@ def test_no_commits(tmp_dir):
     assert Git().no_commits
 
     assert Repo.init().diff() == {}
+
+
+def test_targets(tmp_dir, scm, dvc):
```

Comment:
Can this maybe be broken up into separate tests? 150 lines for a single test is pretty long, and it will be easier to maintain `diff` in general if we can see that we are specifically failing when "`target` is inside a dir out", or "dir out is inside `target`", etc., just by looking at a test case name. As opposed to seeing "we fail inside one of 7 or 8 possible situations when `target` is set".

Reply:
@pmrowla understood, I've attempted to make this change in https://github.com/iterative/dvc/pull/4938/commits/480d951e308a2ba4a9788842b0d0a1180b47d224. Let me know if this was what you expected.